### PR TITLE
extend evict_entry_cache for restart

### LINKED ISSUE #13997: Enhance EntryCache Memory Limiting for TiFlash especially when Restart
## Development Task

Currently, when tiflash cannot keep up with the writing speed of tikv, or TiFlash is restarted, the memory of EntryCache will take up a large overhead. 

In order to reduce the memory usage of EntryCache in TiFlash, we want to strengthen the memory limit of EntryCache for TiFlash.

ref: https://github.com/pingcap/tiflash/issues/6399

### PR DESCRIPTION
**Pull request by hongyunyan**
### What is changed and how it works?

Issue Number: Close #13997

What's Changed:

```commit-message
Support to use evict_entry_cache when restart node.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore/src/store/entry_storage.rs [hunk 1]
```diff
@@ -1227,6 +1227,10 @@ impl<EK: KvEngine, ER: RaftEngine> EntryStorage<EK, ER> {
             let idx = cache.cache[drain_to].index;
             let mem_size_change = cache.compact_to(idx + 1);
             RAFT_ENTRIES_EVICT_BYTES.inc_by(mem_size_change);
+        } else {
```
Comment (BusyJay) [inline components/raftstore/src/store/entry_storage.rs:1230]: Should compact only when half is false.

Comment (hongyunyan) [reply components/raftstore/src/store/entry_storage.rs]: fix

**Review by BusyJay — APPROVED** [review]

**Review by breezewish — APPROVED** [review]

File: components/raftstore/src/store/peer.rs [hunk 2]
```diff
@@ -1188,6 +1188,9 @@ where
             peer.raft_group.campaign()?;
         }
 
+        let persisted_index = peer.raft_group.raft.raft_log.persisted;
+        peer.mut_store().update_cache_persisted(persisted_index);
```
Comment (CalvinNeo) [inline components/raftstore/src/store/peer.rs:1192]: Note: I think this is to update the persisted index in entry cache after restart

**Pull request merged**